narwhal/node/tests/reconfigure.rs

```diff
@@ -175,108 +175,108 @@ async fn run_client(
+//             a.worker_keypairs(),
+//             fixture.worker_cache(),
+//             committee.clone(),
+//             tx_output,
+//             tx_node_reconfigure,
+//         ));
+//
+//         let worker_keypairs = a.worker_keypairs();
+//         let worker_ids = 0..worker_keypairs.len() as u32;
+//         let worker_ids_and_keypairs = worker_ids.zip(worker_keypairs.into_iter()).collect();
+//
+//         let committee = committee.clone();
+//         let worker_cache = worker_cache.clone();
+//
+//         let parameters = Parameters {
+//             batch_size: 200,
+//             max_header_num_of_batches: 1,
+//             ..Parameters::default()
+//         };
+//
+//         let keypair = a.keypair().copy();
+//         let network_keypair = a.network_keypair().copy();
+//         tokio::spawn(async move {
+//             NodeRestarter::watch(
+//                 keypair,
+//                 network_keypair,
+//                 worker_ids_and_keypairs,
+//                 &committee,
+//                 worker_cache,
+//                 /* base_store_path */ test_utils::temp_dir(),
+//                 execution_state,
+//                 parameters,
+//                 TrivialTransactionValidator::default(),
+//                 rx_node_reconfigure,
+//                 &Registry::new(),
+//             )
+//             .await;
+//         });
+//
+//         rx_nodes.push(rx_output);
+//     }
+//
+//     // Give a chance to the nodes to start.
+//     tokio::task::yield_now().await;
+//
+//     // Spawn some clients.
+//     let mut tx_clients = Vec::new();
+//     for a in fixture.authorities() {
+//         let (tx_client_reconfigure, rx_client_reconfigure) = channel(10);
+//         tx_clients.push(tx_client_reconfigure);
+//
+//         let name = a.public_key();
+//         let worker_cache = worker_cache.clone();
+//         tokio::spawn(
+//             async move { run_client(name, worker_cache.clone(), rx_client_reconfigure).await },
+//         );
+//     }
+//
+//     // Listen to the outputs.
+//     let mut handles = Vec::new();
+//     for (tx, mut rx) in tx_clients.into_iter().zip(rx_nodes.into_iter()) {
+//         handles.push(tokio::spawn(async move {
+//             let mut current_epoch = 0u64;
+//
+//             while let Some(epoch) = rx.recv().await {
+//                 println!("Received epoch {}", epoch);
+//                 if epoch == 5 {
+//                     return;
+//                 }
+//                 if epoch > current_epoch {
+//                     current_epoch = epoch;
+//                     tx.send(current_epoch).await.unwrap();
+//                 }
+//             }
+//
+//             if current_epoch < 5 {
+//                 panic!("Node never reached epoch 5, something broke our connection");
+//             }
+//         }));
+//     }
+//
+//     try_join_all(handles)
+//         .await
+//         .expect("No error should occurred");
+// }
```

Comment:
This is to be replaced with a test in the sui repo once the narwhal related validator info is available on chain. Leaving temporarily as commented for reference.

Reply:
I would try to have a simple working version of this test (as we do now have the RegistryService solution in place so nothing blocking us on metrics registration) to at least make sure that the restarting approach works in principle and we don't break things as we go on